src/it/scala/io/iohk/ethereum/sync/util/FakePeer.scala

```diff
@@ -0,0 +1,311 @@
+package io.iohk.ethereum.sync.util
+
+import java.nio.file.Files
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.testkit.TestProbe
+import akka.util.{ByteString, Timeout}
+import io.iohk.ethereum.blockchain.sync.regular.BlockBroadcasterActor
+import io.iohk.ethereum.blockchain.sync.regular.BlockBroadcasterActor.BroadcastBlock
+import io.iohk.ethereum.blockchain.sync.{BlockBroadcast, BlockchainHostActor, TestSyncConfig}
+import io.iohk.ethereum.db.components.{RocksDbDataSourceComponent, Storages}
+import io.iohk.ethereum.db.dataSource.{RocksDbConfig, RocksDbDataSource}
+import io.iohk.ethereum.db.storage.pruning.{ArchivePruning, PruningMode}
+import io.iohk.ethereum.db.storage.{AppStateStorage, Namespaces}
+import io.iohk.ethereum.domain.{Block, Blockchain, BlockchainImpl}
+import io.iohk.ethereum.ledger.InMemoryWorldStateProxy
+import io.iohk.ethereum.mpt.MerklePatriciaTrie
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.PeerManagerActor.{FastSyncHostConfiguration, PeerConfiguration}
+import io.iohk.ethereum.network.discovery.Node
+import io.iohk.ethereum.network.discovery.PeerDiscoveryManager.{DiscoveredNodesInfo, DiscoveryNodeInfo}
+import io.iohk.ethereum.network.handshaker.{EtcHandshaker, EtcHandshakerConfiguration, Handshaker}
+import io.iohk.ethereum.network.p2p.EthereumMessageDecoder
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.rlpx.AuthHandshaker
+import io.iohk.ethereum.network.rlpx.RLPxConnectionHandler.RLPxConfiguration
+import io.iohk.ethereum.network.{EtcPeerManagerActor, ForkResolver, KnownNodesManager, PeerEventBusActor, PeerManagerActor, ServerActor}
+import io.iohk.ethereum.nodebuilder.{PruningConfigBuilder, SecureRandomBuilder}
+import io.iohk.ethereum.sync.util.SyncUtils.{BlockchainState, randomAddress, retryUntilWithDelay}
+import io.iohk.ethereum.utils.ServerStatus.Listening
+import io.iohk.ethereum.utils.{Config, NodeStatus, ServerStatus}
+import io.iohk.ethereum.vm.EvmConfig
+import io.iohk.ethereum.{Fixtures, Timeouts}
+import monix.eval.Task
+
+import scala.concurrent.duration.{FiniteDuration, _}
+
+trait FakePeer extends SecureRandomBuilder with TestSyncConfig {
+  implicit val akkaTimeout: Timeout = Timeout(5.second)
+
+  protected[this] implicit def system: ActorSystem
+
+  def start: Task[Unit]
+
+  val config = Config.config
+
+  import scala.language.postfixOps
+
+  val peerDiscoveryManager = TestProbe().ref
+
+  val nodeKey = io.iohk.ethereum.crypto.generateKeyPair(secureRandom)
+
+  private val nodeStatus =
+    NodeStatus(
+      key = nodeKey,
+      serverStatus = ServerStatus.NotListening,
+      discoveryStatus = ServerStatus.NotListening
+    )
+
+  lazy val tempDir = Files.createTempDirectory("temp-regular-sync")
```

Comment:
Temporary directories in some places are cleared like so:
```scala
  def withDir(testCode: String => Any): Unit = {
    val path = Files.createTempDirectory("testdb").getFileName.toString
    try {
      testCode(path)
    } finally {
      val dir = new File(path)
      assert(!dir.exists() || dir.delete(), "File deletion failed")
    }
  }
```

i wonder if we should do the same here?